Modules/_io/iobase.c

```diff
@@ -671,6 +672,7 @@ _io__IOBase_readlines_impl(PyObject *self, Py_ssize_t hint)
 
     while (1) {
         PyObject *line = PyIter_Next(it);
+        Py_ssize_t line_length;
```

Comment:
Better move this line before the above. IIRC C89 doesn't allow mixing variable declaration and assignment.